### DISCUSSION TITLE
NPC and Trainer Profession based trades

### DISCUSSION THREAD

**Discussion by Thutmose** [Pokecube-Development/Pokecube-Issues-and-Wiki#841] [Ideas]
### healer (pokecenter block)
1. - 8 revives for 4 emeralds
    - 16 berry juice for 4 emeralds
2.  - 4 vitamins for 12 emeralds (random vitamins selected)
     - Map to nearest town for 4 emeralds
     - Map to nearest village for 4 emeralds
3.  - 4 berries for 3 emeralds (random berry)
    - 1 emerald for 8 berries (random berry)
4.  - 3 emerald for 3 glistening melon
    - 3 emerald for 3 golden carrot
5.  - 2 emerald for 1 golden apple

### professor (genetics blocks)
1. - 1 pokedex/pokewatch for 16 emeralds (1 time trade each, gets it to next level)
2. - 1 exp share for 16 emeralds
3. - 1 pokebag for 16 emeralds
   - 1 energy siphon for 64 emeralds
4. - Pokecube Sales*
5. - 1 cloning device for 64 emeralds
   - 1 gene extractor for 96 emeralds
   - Map to the nearest Elite Four

### trader (trading table)
1. - 1 emerald for 8 berries (random berry)
2. - Pokecube Sales*
3. - 4 berries for 3 emeralds (random berry)
4. - mega evo item sales
    - normal evo item sales
5.  - shiny charm sale
     - legendary item sales for 16,32,64 emeralds

*Pokecube Sales are:
16 pokecubes for 4 emeralds
8 greatcubes for 4 emeralds
4 ultracubes for 4 emeralds

### Sailor

1.  - Map to nearest ocean village
2.  - Map to nearest lugia tower
3.  - Map to nearest kyogre temple
4.  - Map to nearest tapu temple
5.  - Map to nearest sky pillar
     - Map to nearest Kubfu Water

### Fisherman

1. - 4 lurecubes for 4 emeralds
2. - 4 netcubes for 4 emeralds
3. - Fishing rod for 4 emeralds
    - 8 raw cod for 1 emerald
    - 8 raw salmon for 1 emerald
4. - Map to nearest Nature Place
5. - Map to nearest Zazamenta Temple
   - Map to nearest Zacian Temple

### Hiker

1. - 1 small bag for 4 emeralds
    - 1 compass for 4 emeralds
2. - 4 berries for 2 emeralds (random berry)
3. - 1 emerald for 3 bread
4. - 1 emerald for 4 cookies
5. - Map to nearest Keldeo Place
    - Map to nearest Kubfu Dark

### Scientist

1. - 4 empty pokemob eggs for 1 emerald
2. - 1 Writable book for 8 emeralds
    - 1 clock for 8 emeralds
3. - 2 blank TMs for 1 emerald
4. - Map to nearest N Castle
5. - Map to nearest Team Rocket Hideout
    - Map to nearest Team Magma Hideout
    - Map to nearest Team Aqua Hideout

### Pokemaniac

1. - 4 repeatcubes for 1 emerald
    - 4 timercubes for 1 emerald
2. - 4 berries for 3 emeralds (random berry)
3. - Map to nearest Unown Temple
4. - Map to nearest Regi Temple (any except regigigas
5. - Map to nearest Regigigas Temple

### Black Belt

1. - 1 belt for 5 emeralds
    - 1 bracelet for 3 emeralds
2. - 1 emerald for 4 berries (random berry)
3. - 1 mega belt for 5 emeralds
4. - Map to nearest Kubfu Dojo
5. - Map to nearest Kubfu Water
    - Map to nearest Kubfu Dark

### Bug Catcher

1. - 4 netcubes for 1 emerald
2. - 4 nestcubes for 1 emerald
3. - 4 string for 1 emerald
    - 4 spider eyes for 1 emerald
4. - 1 emerald for 3 bread
5. - Map to nearest Celebi Temple
    - Map to nearest Xerneas Place

### Camper

1. - 1 small bag for 4 emeralds
    - 1 compass for 4 emeralds
2. - 4 berries for 2 emeralds (random berry)
3. - 1 emerald for 3 bread
4. - 1 emerald for 4 cookies
5. - Map to nearest Ho-oh tower
    - Map to nearest Legendy Tree
    - Map to nearest Yveltal Place

## List of Structures, and who sells maps to them

| Structure                       | Seller        |
| ------------------------- | ----------- |
| Lugia Tower                  | Sailor         |
| Kyogre Temple              | Sailor        |
| Tapu Temple                 | Sailor        |
| Sky Pillar                       | Sailor        |
| Kubfu Dojo                   | Black Belt |
| Kubfu Water                 | Sailor, Black Belt |
| Kubfu Dark                   | Hiker, Black Belt  |
| Keldeo Place                 | Hiker  |
| Nature Place                 | Fisherman |
| Zacian Temple              | Fisherman |
| Zazamenta Temple       | Fisherman |
| Regirock Temple           | Pokemaniac  |
| RegiceTemple               | Pokemaniac |
| Registeel Temple          | Pokemaniac |
| Regidrago Temple        | Pokemaniac |
| Regieleki Temple          | Pokemaniac |
| Regigigas Temple         | Pokemaniac |
| Yveltal Place                 | Camper |
| Legendy Tree               | Camper |
| Celebi Temple              | Bug Catcher |
| Space Temple  |  |
| Ho-oh Tower               | Camper |
| Xerneas Place              | Bug Catcher |
| Groudon Temple  |  |
| Necrozma Tower  |  |
| N Castle  |  Scientist |
| Elite Four  | Professor (Other than Spawn one) |
| Team Rocket Hideout |  Scientist  |
| Team Magma Hideout |  Scientist  |
| Team Aqua Hideout |  Scientist  |
| Unown Temple | Pokemaniac |

**Comment by Thutmose** [comment]
Any suggestions, etc for who should sell what would be appreciated

**Comment by TheSignalist** [comment]
Hey @Thutmose,

It would be really useful if the trades could be configured through a config.

Just so you have an example, I'm currently trying to build a modpack around pokecube, and I've modified the recipes for pokeballs to better incorporate with the rest of mods (Create and Tinkers, main mentions for now). And I've been trying to find a way to modify the trades as I'd like to remove certain pokecubes from the list, as well as modify prices of other items.

Btw, if you have an "easy" way to do this through KubeJS/Crafttweaker/Datapacks, please let me know, it would help me a lot, as I've not been able to find anything and I'm starting to go crazy xD.

Finally, just wanted to thank you for making and maintaining this awesome mod, it is one of my favorite ones (and I'm not even a big pokemon fan xD)
